ext/intl/transliterator/transliterator_class.c

```diff
@@ -144,17 +144,13 @@ static zend_object *Transliterator_clone_obj( zend_object *object )
 
 	if( to_orig->utrans != NULL )
 	{
-		zval			tempz; /* dummy zval to pass to transliterator_object_construct */
-
 		/* guaranteed to return NULL if it fails */
 		UTransliterator *utrans = utrans_clone( to_orig->utrans, TRANSLITERATOR_ERROR_CODE_P( to_orig ) );
 
 		if( U_FAILURE( TRANSLITERATOR_ERROR_CODE( to_orig ) ) )
 			goto err;
 
-		ZVAL_OBJ(&tempz, ret_val);
-		transliterator_object_construct( &tempz, utrans,
-			TRANSLITERATOR_ERROR_CODE_P( to_orig ) );
+		to_new->utrans = utrans;
 
 		if( U_FAILURE( TRANSLITERATOR_ERROR_CODE( to_orig ) ) )
```

Comment:
There is no more need for the duplicate error checking. The one before the assignment can be removed, as well as the `err:` label a few lines below.

Reply:
I updated the code, did I get it right?